src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -332,115 +357,99 @@ def _post_process(
+    print(f"Start uploading {assembly_files_len} assembly files\n")
 
-    # Put the assembly files in the input_queue
-    counter = 1
-    for assembly_name, host_assembly_dir in wait_to_upload_assemblies.items():
+    uploaded_count = 0
+    uploaded_fail = False
+    for assembly_tuples in _gen(wait_to_upload_assemblies, batch_size):
 
-        container_internal_assembly_path = os.path.join(
-            JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER, assembly_name
-        )
-        conf.input_queue.put(
-            (
+        try:
+            batch_upas = _upload_assemblies_to_workspace(asu, workspace_id, assembly_tuples)
+            batch_uploaded_tuples = assembly_tuples
+        except Exception as e:
+            name_assemblyTuple_map = {
+                assembly_tuple.assembly_name: assembly_tuple for assembly_tuple in assembly_tuples
+            }
+            assembly_objects_in_ws = loader_helper.list_objects(
+                workspace_id,
+                ws,
+                loader_common_names.OBJECTS_NAME_ASSEMBLY,
+            )
+            print("assembly_objects_in_ws: ", assembly_objects_in_ws)
+            name_upa_map = {
+                object_info[1]: "{6}_{0}_{4}".format(*object_info) for object_info in assembly_objects_in_ws
+            }
+            batch_uploaded_objects_names = set(name_upa_map.keys()) & set(name_assemblyTuple_map.keys())
+
+            # update batch_uploaded_tuples and batch_upas
+            batch_upas = tuple([name_upa_map[name] for name in batch_uploaded_objects_names])
+            batch_uploaded_tuples = [name_assemblyTuple_map[name] for name in batch_uploaded_objects_names]
+            uploaded_fail = True
+
+        for assembly_tuple, upa in zip(batch_uploaded_tuples, batch_upas):
+            _post_process(
                 upload_env_key,
                 workspace_id,
-                container_internal_assembly_path,
-                host_assembly_dir,
-                assembly_name,
+                assembly_tuple.host_assembly_dir,
+                assembly_tuple.assembly_name,
                 upload_dir,
-                counter,
-                assembly_files_len,
+                output_dir,
+                upa
             )
-        )
 
-        if counter % 5000 == 0:
-            print(f"Jobs added to the queue: {counter}/{assembly_files_len}, "
-                  f"Percentage: {counter / assembly_files_len * 100:.2f}%, "
+        uploaded_count += len(batch_uploaded_tuples)
+        if uploaded_count % 100 == 0:
+            print(f"Assemblies uploaded: {uploaded_count}/{assembly_files_len}, "
+                  f"Percentage: {uploaded_count / assembly_files_len * 100:.2f}%, "
                   f"Time: {datetime.now()}")
 
-        counter += 1
-
-    # Signal the workers to terminate when they finish uploading assembly files
-    for _ in range(conf.workers):
-        conf.input_queue.put(None)
+        if uploaded_fail:
+            return uploaded_count
```

Comment:
So if I understand correctly, on fail the uploader 
1) Figures out which uploads succeeded
2) Does post processing on those uploads
3) Halts and prints out the number of successful uploads.

What is the user expected to do at that point?

Reply:
Very close, on fail the uploader
1. print out error message (this step is in _upload_assemblies_to_workspace func)
2. Figures out which uploads succeeded
3. Does post processing on those uploads
4. Existed and still calculate upload speed on successful ones

The user is expected to look at that particular assembly/object from error messages/logs and figure out why it causes the failure. For example,  illegal object name. Once fit, rerun with the same command.

Reply:
And it figures out where it left off from the yaml files?

Reply:
right, will skip successful ones in the next run.

Reply:
Ok. We might find that to save time the script should restart by itself and summarize errors at the end, but we can add that later

Reply:
: